packages/repco-webcomponent/src/repco-post-element/repco-post-element.ts

```diff
@@ -0,0 +1,155 @@
+      color: var(--dark-card-color, #f7fafc);
+      box-shadow: var(--dark-card-shadow, 0px 4px 6px rgba(0, 0, 0, 0.1));
+      border-color: var(--dark-card-border-color, #cbd5e0);
+    }
+    :host([theme='dark']) .card:hover {
+      background-color: var(--dark-card-hover-background-color, #1c1c1c);
+      box-shadow: var(--dark-card-hover-shadow, 0px 4px 6px rgba(0, 0, 0, 0.1));
+      border-color: var(--dark-card-hover-border-color, #cbd5e0);
+    }
+  `
+  @property({ type: String, reflect: true })
+  theme = 'light'
+
+  @property({ type: String })
+  layout = 'column'
+
+  @property({ type: Number })
+  count = 10
+
+  @property({ type: String })
+  _defaultThumbnail = ''
+
+  @property()
+  endpoint = 'https://node1.repco.openaudiosearch.org/graphql'
+
+  @property({ type: String })
+  query = `
+  query {
+    contentItems {
+      nodes {
+        title
+        content
+        uid
+        mediaAssets {
+          nodes {
+            mediaType
+            file {
+              contentUrl
+            }
+          }
+        }
+      }
+    }
+  }
+`
+  @state()
+  private _state: { posts: PostType[] } = { posts: [] }
+
+  private _repcoFetchClient: any
+
+  override async firstUpdated() {
+    this.fetchData()
+  }
+
+  override render() {
+    return html` <repco-fetch-api
+        .endpoint="${this.endpoint}"
+        .query="${this.query}"
+      >
+      </repco-fetch-api>
+      <div
+        style="display: flex; flex-direction: ${this.layout === 'horizontal'
+          ? 'row'
+          : 'column'};"
+      >
+        ${this._state.posts.slice(0, this.count).map((post) => {
+          let thumbnail = this._defaultThumbnail
+          post.mediaAssets.nodes.map((asset: MediaAssetType) => {
+            if (asset.mediaType == 'image') {
+              thumbnail = asset.file.contentUrl
+            }
+          })
+
+          const header = post.title
+          const subheader = post.uid
+
+          const body = `${this.trimContent(post.content)}...`
+          const footer = `source: ${this.endpoint}`
+          const endpointBase = this.endpoint.substring(
+            0,
+            this.endpoint.lastIndexOf('/'),
+          )
+          const link = `${endpointBase}/items/${post.uid}`
+
+          return html`
+            <repco-post-card
+              class="card"
+              .thumbnail="${thumbnail}"
+              .header="${header}"
+              .subheader="${subheader}"
+              .body="${body}"
+              .footer="${footer}"
+              @click=${() => this.cardClick(link)}
+            ></repco-post-card>
+          `
+        })}
+      </div>`
+  }
+
+  private async fetchData() {
+    if (this.shadowRoot) {
+      this._repcoFetchClient =
+        this.shadowRoot.querySelector('repco-fetch-client')
+
+      if (this._repcoFetchClient) {
+        await this._repcoFetchClient.updateComplete
+        this._repcoFetchClient.addEventListener(
```

Comment:
Yes, exactly the idea is to gain more flexibility. For example, apollo provides a component in the apollo-elements where you can also use the caching of the apollo client. so I wanted to keep it open and make the way the data comes into the repco-post-element or into the repco-post-card as flexible as possible. for now i think the way via fetch is good but if the requirements for the webcomponent become more demanding we have a good basis to adapt here.

The approach with the event listner also offers a good possibility to adapt the content of the web component more flexibly or on demand when new data comes in or when it is triggered that new content should be displayed.

Reply:
I see. I'm still not sure how this would work at the moment. How could these components be plugged together differently by users?

Reply:
At the moment it does not bring anything to the user for that we need another abstraction. It is more for us. At the moment I would make the repco-post-element available to the user which contains the repco-fetch-api. at a later point in time you could rebuild this so that you pass an object into the post element which is rendered on the repco-post-card. this would then allow the user to determine how he wants to pass the data into the web component whether with the fetch, apollo-ellement provided by us or because of me also as a static object (which of course would be gross nonsense).